### ISSUE TITLE
pub test sometimes doesn't terminate on second Ctrl+C

### ISSUE THREAD

**Issue by DanTup** [dart-lang/test#1114]
I was running some Flutter tests (on macOS) with:

```
pub run test test/general.shard/
```

I no longer needed the results, so I hit `Ctrl+C` and then when it told me to press it again to terminate immediately, I pressed it again. However the process didn't terminate. I pressed it a few more times, but several minutes later it's still alive:

<img width="596" alt="Screenshot 2019-11-26 at 3 02 04 pm" src="https://user-images.githubusercontent.com/1078012/69645224-bbfacc00-105d-11ea-8fe2-263f21b99c1d.png">

I'm not sure what might have happened for it not to quit. I expected it to terminate even if it didn't clean up nicely.

**Comment by jonasfj** [comment]
I think the "waiting for tests to finish" is a `package:test` message.

**Comment by grouma** [comment]
This will be hard to track down without more details. I haven't experienced this myself but it may be a Windows only issue?

For some details:
`package:test` sets up some listeners to intercept shutdown signals:
https://github.com/dart-lang/test/blob/17850e02e6359ce9fc9349aa769521a2e84b22c8/pkgs/test_core/lib/src/executable.dart#L65

We intercept them and do a graceful shutdown here:
https://github.com/dart-lang/test/blob/17850e02e6359ce9fc9349aa769521a2e84b22c8/pkgs/test_core/lib/src/executable.dart#L143

After the first intercept, we cancel all listeners so subsequent shutdown requests should use the default handler. I'm wondering if that's not the case for Windows.

**Comment by DanTup** [comment]
That was on my corp MacBook, so I don't think it's that. I'll have another go at reproducing it running the same tests tomorrow. When I tried to repro I was only running a subset (since the whole run is a bit slow), but it's possible there's something weird going on in the other test code it's running.

**Comment by natebosch** [comment]
To confirm - this was with `pub run test` and not `pub run build_runner test` correct?

I know we have had this issue with the latter command, I'm not sure if we have had it reported with `pub run test`.

**Comment by DanTup** [comment]
Yeah, it was definitely `pub run test test/general.shard/` (to be sure I grep'd my bash_history and that's all that's there for running this `genera.shard` folder).
